### PR TITLE
fix(market): Handle empty agents response

### LINKED ISSUE #8370: [Frontend/Market] Graceful handling when no agents or featured agents
When there are no agents or featured agents returned from the backend, the marketplace breaks and won't display anything. We should handle this more gracefully<br><br>374-caa09da3d09594f2.js:14 TypeError: Cannot read properties of undefined (reading 'length') at j (page-c8cc863c15f77079.js:1:9859) at rE (fd9d1056-752c9b863fd314ce.js:1:40686) at l$ (fd9d1056-752c9b863fd314ce.js:1:59661) at iZ (fd9d1056-752c9b863fd314ce.js:1:118268) at ia (fd9d1056-752c9b863fd314ce.js:1:95507) at fd9d1056-752c9b863fd314ce.js:1:95329 at il (fd9d1056-752c9b863fd314ce.js:1:95336) at oJ (fd9d1056-752c9b863fd314ce.js:1:92692) at oZ (fd9d1056-752c9b863fd314ce.js:1:92111) at MessagePort.j (374-caa09da3d09594f2.js:1:108919)

### PR DESCRIPTION
**Pull request by Abhi1992002**
### Fixes #8370 

I added a check to handle cases where no agents or featured agents are returned from the backend. This prevents the marketplace from crashing due to an undefined property error and ensures the UI still loads properly.

### Changes 🏗️

Added fallback messages for when featuredAgents and topAgents are empty.

Screenshot 2024-10-22 at 6 45 45 PM

### CODE REVIEW

File: autogpt_platform/supabase
Comment (ntindle) [inline autogpt_platform/supabase:1]: revert the supabase change and this should be good to go

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt_platform/supabase
Comment (Abhi1992002) [reply autogpt_platform/supabase:1]: Done..!!

**Comment by aarushik93** [comment]
@Abhi1992002 thanks for this, but the supabase submodule shouldn't be deleted in this PR. Can you please fix that?

**Review by aarushik93 — APPROVED** [review]
LGTM

**Pull request merged**